spring-integration-core/src/main/java/org/springframework/integration/support/json/Jackson3JsonMessageParser.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2025-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.json;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.jspecify.annotations.Nullable;
+import tools.jackson.core.JacksonException;
+import tools.jackson.core.JsonParser;
+import tools.jackson.core.JsonToken;
+import tools.jackson.core.ObjectReadContext;
+import tools.jackson.core.json.JsonFactory;
+
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * {@link JsonInboundMessageMapper.JsonMessageParser} implementation that parses JSON messages
+ * and builds a {@link Message} with the specified payload type from provided {@link JsonInboundMessageMapper}.
+ * Uses <a href="https://github.com/FasterXML">Jackson JSON Processor</a>.
+ *
+ * @author Jooyoung Pyoung
+ *
+ * @since 7.0
+ */
+public class Jackson3JsonMessageParser extends AbstractJacksonJsonMessageParser<JsonParser> {
+	private static final JsonFactory JSON_FACTORY = JsonFactory.builder().build();
+
+	public Jackson3JsonMessageParser() {
+		this(new Jackson3JsonObjectMapper());
+	}
+
+	public Jackson3JsonMessageParser(Jackson3JsonObjectMapper objectMapper) {
+		super(objectMapper);
+	}
+
+	@Override
+	protected JsonParser createJsonParser(String jsonMessage) throws JacksonException {
```

Comment:
You said this `JacksonException`  is an unchecked exception, so no need to mention it here I believe.